internal/logger/target/http/http.go

```diff
@@ -309,19 +318,69 @@ func (h *Target) startHTTPLogger(ctx context.Context) {
 		return
 	}
 
+	buf := bytebufferpool.Get()
+	defer bytebufferpool.Put(buf)
+
+	json := jsoniter.ConfigCompatibleWithStandardLibrary
+	enc := json.NewEncoder(buf)
+	batchSize := h.batchSize
+	if batchSize <= 0 {
+		batchSize = 1
+	}
+
+	payloadType := "application/json"
+	if batchSize > 1 {
+		payloadType = ""
+	}
+
+	timer := time.NewTimer(3 * time.Second)
+	defer timer.Stop()
+
+	var nevents int
 	// Send messages until channel is closed.
-	for entry := range logCh {
-		h.logEntry(ctx, entry)
+	for {
+		select {
+		case entry, ok := <-logCh:
+			if !ok {
+				return
+			}
+			atomic.AddInt64(&h.totalMessages, 1)
+			nevents++
+			if err := enc.Encode(&entry); err != nil {
+				atomic.AddInt64(&h.failedMessages, 1)
+				continue
+			}
+			if nevents == batchSize {
+				h.logEntry(ctx, buf.Bytes(), payloadType)
+				buf.Reset()
+				nevents = 0
+			}
+		case <-timer.C:
+			// Timer trigger to flush any pending events in the `buf.Bytes()`
+			// when there are no incoming events on logCh we should flush out
+			// the entries.
+			if len(logCh) > 0 {
+				timer.Reset(3 * time.Second)
+				continue
+			}
```

Comment:
```suggestion
```

Not needed if we add the above, and we risk having messages "stuck" for quite a while.